connectors/sources/sharepoint_online.py

```diff
@@ -929,6 +942,13 @@ def is_domain_group(user_fields):
     )
 
 
+def is_dynamic_group(login_name):
+    if not login_name:
+        return False
+
+    return login_name.startswith("c:0o.c|federateddirectoryclaimprovider")
```

Comment:
nit: should we stay consistent when using `startswith()` and include `|` at the end of the string pattern (same as in `_get_login_name`)? so it would be:
```
login_name.startswith("c:0o.c|federateddirectoryclaimprovider")
```